corehq/apps/users/models.py

```diff
@@ -1214,6 +1241,29 @@ def user_session_data(self):
             '{}_last_name'.format(SYSTEM_PREFIX): self.last_name,
             '{}_phone_number'.format(SYSTEM_PREFIX): self.phone_number,
         })
+
+        domain = None
+        try:
+            domain = self.domain
+        except AttributeError:
+            domains = self.get_domains()
+            if domains:
+                domain = domains[0]
+
+        if domain is not None:
+            session_data['{}_project'.format(SYSTEM_PREFIX)] = domain
+
+        if self.location_id:
+            session_data.update({
+                '{}_location_id'.format(SYSTEM_PREFIX): self.location_id,
+                '{}_primary_case_sharing_id'.format(SYSTEM_PREFIX): self.location_id,
+            }
+            )
+        if self.assigned_location_ids:
+            session_data['{}_location_ids'.format(SYSTEM_PREFIX)] = user_location_data(self.assigned_location_ids)
+        if self.commtrack_supply_point:
+            session_data['commtrack-supply-point'] = self.commtrack_supply_point
```

Comment:
Do you mean this is an incorrect change (i.e. it shouldn't be applied to web users), or is just confusing to read because it only applies to mobile workers?

Reply:
Sorry, yeah, that was unclear.  I mean that I think this is a behavior change.  That is, in the current state of the code, the location-related fields are all added to the metadata only for mobile workers.  For example, `WebUser.set_location` doesn't modify `self.metadata`, but `CommCareUser.set_location` does.

This shouldn't actually matter, because `WebUser` locations are set on the domain membership, but due to some accident of history, `location_id` and `assigned_location_ids` are set on `CouchUser` instead of `CommCareUser`, so there's probably some data out there with those fields set also for WebUsers.

Regarding correctness, I think putting any domain-specific data in `WebUser.user_session_data` (which is _not_ domain specific) is incorrect, as per the comment thread above, but I do agree it's safer for this PR's scope to keep that behavior consistent, as you've done with `commcare_project`.